elastic/apm-server
# Generate json schema

## Motivation/summary
This PR removes the manually crafted JSON schemas for the Intake API. 

Previous refactoring of the Intake decoder made JSON schemas obsolete in the decoder process. They are still kept around as they are an integral part of the APM agent development cycle, where automated tests are run against the schemas. To ensure the JSON schemas do not get out of sync with the actual decoder logic, this PR introduces a code generator that builds the JSON schemas, based on the internally used go structs, their validation tags and comments. 

Whenever new fields are added running `make update` will re-generate the JSON schemas and keep them in sync. 

## Checklist

~- [ ] I have signed the [Contributor License Agreement](https://www.elastic.co/contributor-agreement/).~

I have considered changes for:
~- [ ] logging (add log lines, choose appropriate log selector, etc.)~
~- [ ] metrics and monitoring (create issue for Kibana team to add metrics to visualizations, e.g. [Kibana#44001](https://github.com/elastic/kibana/issues/44001))~
~- [ ] telemetry~
~- [ ] Elasticsearch Service (https://cloud.elastic.co)~
~- [ ] Elastic Cloud Enterprise (https://www.elastic.co/products/ece)~
~- [ ] Elastic Cloud on Kubernetes (https://www.elastic.co/elastic-cloud-kubernetes)~

## How to test these changes
Add/remove/update fields, comments and implemented validation tags and run `make update` and verify the changes are accordingly reflected in the JSON schemas. 

## Related issues
#4334
